tests/test_efadv.py

```diff
@@ -27,6 +28,7 @@ def test_efadv_query(self):
         """
         Verify that it's possible to read EFA direct-verbs.
         """
+        efa_base.skip_if_not_efa_dev(self.ctx)
```

Comment:
I prefer we add a setUp function that verifies the device (like you did in test_efa_srd) instead of coding this in every test case.

Reply:
Do you mean add setUp functions to all test classes under test_efadv.py? 

Or, I think it will be better to create two new class "EfaAPITestCase" and "EfaRDMATestCase" under efa_base.py and change all the efa tests to inhirent from them + make sure to implement the setUp function will the check of device/vendor id for skipping the test if the device is not EFA, thoughts?

Reply:
I can't see the old version now, but I think you added the EFA device check to the setUp() function in QPSRDTestCase, right?

Why not add a setUp function to EfaQueryDeviceTest as well and perform the check there, instead of in each test case.

Reply:
@galpress Please review the latest changes, I think that this approach is much cleaner.